File: 品味Java的21種設計模式/dp_java/state/example7/AuditOverState.java
package cn.javass.dp.state.example7;

import java.util.Scanner;
/**
 * 处理审核结束的类
 */
public class AuditOverState implements LeaveRequestState{
	public void doWork(StateMachine request) {
		//先把业务对象造型回来
		LeaveRequestModel lrm = (LeaveRequestModel)request.getBusinessVO();

		//业务处理，在数据里面记录整个流程结束		
	}
}

File: 品味Java的21種設計模式/dp_cpp/state/example7/AuditOverState.cpp
#include "AuditOverState.h"

namespace cn {
namespace javass {
namespace dp {
namespace state {
namespace example7 {

void AuditOverState::doWork(StateMachine *request) {
    //先把业务对象造型回来
    LeaveRequestModel *lrm = static_cast<LeaveRequestModel *>(request->getBusinessVO());

    //业务处理，在数据里面记录整个流程结束
}
}
}
}
}
}
